types/block.go

```diff
@@ -988,29 +1002,105 @@ func CommitFromProto(cp *tmproto.Commit) (*Commit, error) {
 
 //-----------------------------------------------------------------------------
 
-// Data contains the set of transactions included in the block
+// Data contains all the available Data of the block.
+// Data with reserved namespaces (Txs, IntermediateStateRoots, Evidence) and
+// Celestia application specific Messages.
 type Data struct {
-
 	// Txs that will be applied by state @ block.Height+1.
 	// NOTE: not all txs here are valid.  We're just agreeing on the order first.
 	// This means that block.AppHash does not include these txs.
 	Txs Txs `json:"txs"`
 
+	Evidence EvidenceData `json:"evidence"`
+
+	// The blobs included in this block.
+	Blobs []Blob `json:"blobs"`
+
+	// SquareSize is the size of the square after splitting all the block data
+	// into shares. The erasure data is discarded after generation, and keeping this
+	// value avoids unnecessarily regenerating all of the shares when returning
+	// proofs that some element was included in the block
+	SquareSize uint64 `json:"square_size"`
+
 	// Volatile
 	hash tmbytes.HexBytes
 }
 
-// Hash returns the hash of the data
+// Hash returns the hash of the data. `data.hash` is expected to be set by
+// PrepareProposal in celestia-app. However, this function falls back to
+// calculating `data.hash` as vanilla tendermint would in order to satisfy unit
+// and e2e tests.
 func (data *Data) Hash() tmbytes.HexBytes {
 	if data == nil {
 		return (Txs{}).Hash()
 	}
 	if data.hash == nil {
 		data.hash = data.Txs.Hash() // NOTE: leaves of merkle tree are TxIDs
 	}
+
+	// this is the expected behavior where `data.hash` was set by celestia-app
+	// in PrepareProposal
 	return data.hash
```

Comment:
Note: this is where the `data.hash` set by celestia-app is propagated to the Tendermint block's data hash